.github/workflows/php.yml

```diff
@@ -0,0 +1,52 @@
+name: PHP Composer
+
+on:
+  push:
+    branches: [ 2.x ]
+  pull_request:
+    branches: [ 2.x ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    
+    strategy: 
+      fail-fast: false
+      matrix: 
+        php-versions: ['7.1','7.2','7.3', '7.4','8.0']
+        phpunit-versions: ['latest']
+      
+    steps:
+    - uses: actions/checkout@v2
+    - uses: shivammathur/setup-php@v2
+      with:
+        php-version: ${{ matrix.php-versions }} #'7.2'
+        tools: composer:v1
+        extensions: mbstring, intl
+        #ini-values: pcov.directory=api #optional, see above for usage.
+        coverage: pcov
+     
+    - name: Validate composer.json and composer.lock
+      run: composer validate
+
+    - name: Cache Composer packages
+      id: composer-cache
+      uses: actions/cache@v2
+      with:
+        path: vendor
+        key: ${{ runner.os }}-php-${{ hashFiles('**/composer.json') }}
```

Comment:
The cache key should contain the date because otherwise the same cache entry is being used until there is a change in composer.json. (This is not just a blocker for daily builds, see above)